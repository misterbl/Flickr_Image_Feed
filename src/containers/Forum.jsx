import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { getForumData } from "../selectors/apiSelectors";
import ForumCard from "../components/ForumCard";
import * as apiThunk from "../actions/thunks/apiThunk";

export class Forum extends React.Component {
  render() {
    const { forumData } = this.props;
    if (forumData) {
      return (
        <>
          <h1 className="tc">{forumData.title}</h1>
          <div className="ma3">
            {forumData &&
              forumData.items.map(item => (
                <ForumCard
                  title={item.title}
                  author={item.author}
                  author_id={item.author_id}
                  link={item.link}
                  description={item.description}
                  key={item.title + item.published}
                />
              ))}
          </div>
        </>
      );
    }
    this.props.getForumFeed();
    return <h1 className="tc">Fetching data...</h1>;
  }
}

export const mapStateToProps = state => ({
  forumData: getForumData(state)
});

export const mapDispatchToProps = {
  getForumFeed: apiThunk.getForumFeed
};
export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Forum)
);
