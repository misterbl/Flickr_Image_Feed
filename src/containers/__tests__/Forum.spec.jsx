import * as React from "react";
import { shallow } from "enzyme";
import { Forum } from "../Forum";
import generateAppState from "../../testMocks/appState.mock";

describe("Forum", () => {
  const state = generateAppState();
  const props = {
    forumData: state.api.feed.flickrForum,
    getForumFeed: jest.fn()
  };

  const wrapper = shallow(<Forum {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders the ForumCard if ForumData is not null", () => {
    const forumCard = wrapper.find("ForumCard");
    expect(forumCard.length).toEqual(1);
  });
  it("doesn't render the ForumCard but a h1 tag if ForumData is null", () => {
    const newProps = { ...props, forumData: null };
    const newWrapper = shallow(<Forum {...newProps} />);
    const forumCard = newWrapper.find("ForumCard");
    const h1 = newWrapper.find("h1");
    expect(forumCard.length).toEqual(0);
    expect(h1.length).toEqual(1);
  });
  describe("conponentDidMount", () => {
    it("calls getForumFeed", () => {
      wrapper.instance();
      expect(props.getForumFeed).toHaveBeenCalled();
    });
  });
});
