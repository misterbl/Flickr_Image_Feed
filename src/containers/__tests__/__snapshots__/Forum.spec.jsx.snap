// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forum matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Forum
    forumData={
      Object {
        "items": Array [
          Object {
            "author": "authorTest",
            "author_id": "123",
            "description": "descriptionTest",
            "link": "linkTest",
            "title": "titleTest",
          },
        ],
        "title": "Flickr forum",
      }
    }
    getForumFeed={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="tc"
        >
          Flickr forum
        </h1>,
        <div
          className="ml6"
        >
          <ForumCard
            author="authorTest"
            author_id="123"
            description="descriptionTest"
            link="linkTest"
            title="titleTest"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Flickr forum",
          "className": "tc",
        },
        "ref": null,
        "rendered": "Flickr forum",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForumCard
              author="authorTest"
              author_id="123"
              description="descriptionTest"
              link="linkTest"
              title="titleTest"
            />,
          ],
          "className": "ml6",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "titleTestundefined",
            "nodeType": "function",
            "props": Object {
              "author": "authorTest",
              "author_id": "123",
              "description": "descriptionTest",
              "link": "linkTest",
              "title": "titleTest",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="tc"
          >
            Flickr forum
          </h1>,
          <div
            className="ml6"
          >
            <ForumCard
              author="authorTest"
              author_id="123"
              description="descriptionTest"
              link="linkTest"
              title="titleTest"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Flickr forum",
            "className": "tc",
          },
          "ref": null,
          "rendered": "Flickr forum",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForumCard
                author="authorTest"
                author_id="123"
                description="descriptionTest"
                link="linkTest"
                title="titleTest"
              />,
            ],
            "className": "ml6",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "titleTestundefined",
              "nodeType": "function",
              "props": Object {
                "author": "authorTest",
                "author_id": "123",
                "description": "descriptionTest",
                "link": "linkTest",
                "title": "titleTest",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
